src/index.js

```diff
@@ -117,8 +118,17 @@ export default class VueRouter {
     const history = this.history
 
     if (history instanceof HTML5History || history instanceof HashHistory) {
-      const setupListeners = () => {
+      const handleInitialScroll = (route) => {
+        const expectScroll = this.options.scrollBehavior
+        const supportsScroll = supportsPushState && expectScroll
+
+        if (supportsScroll) {
+          handleScroll(this, route, route, false)
```

Comment:
I think it would make sense to pass the initial location as well so get the initial location outside of `handleInitialScroll`

```js
const from = history.current
```

and then do

```js
handleScroll(this, route, from, false)
```

Reply:
make sense, changed in new commit